File: table-service/src/modules/table/table.controller.ts
import { Controller, Get, Post, Patch, Param, Body, HttpCode, HttpStatus } from '@nestjs/common';
import { ApiTags, ApiOperation, ApiResponse, ApiParam, ApiBody } from '@nestjs/swagger';
import { TableService } from './table.service';
import { CreateTableDto } from './dto/create-table.dto';
import { UpdateTableStatusDto } from './dto/update-table-status.dto';

@ApiTags('Tables')
@Controller('tables')
export class TableController {
  constructor(private readonly tableService: TableService) {}

  @Post()
  @ApiOperation({ summary: 'Create new table with QR code' })
  @ApiResponse({ status: 201, description: 'Table created.' })
  @HttpCode(HttpStatus.CREATED)
  create(@Body() createTableDto: CreateTableDto) {
    return this.tableService.create(createTableDto);
  }

  @Get()
  @ApiOperation({ summary: 'Get all tables' })
  @ApiResponse({ status: 200, description: 'Tables list.' })
  findAll() {
    return this.tableService.findAll();
  }

  @Patch(':id/status')
  @ApiOperation({ summary: 'Update table status' })
  @ApiParam({ name: 'id', type: String })
  @ApiBody({ type: UpdateTableStatusDto })
  updateStatus(@Param('id') id: string, @Body() updateStatusDto: UpdateTableStatusDto) {
    return this.tableService.updateStatus(id, updateStatusDto.status);
  }

  @Post(':id/call-staff')
  @ApiOperation({ summary: 'Call staff for table' })
  @ApiParam({ name: 'id', type: String })
  @ApiResponse({ status: 200, description: 'Staff called.' })
  @HttpCode(HttpStatus.OK)
  callStaff(@Param('id') id: string) {
    return this.tableService.callStaff(id);
  }
}

File: apps/backend/table-service/src/main/java/com/coffeeshop/tableservice/controller/TableController.java
package com.coffeeshop.tableservice.controller;

import com.coffeeshop.tableservice.dto.CallStaffRequest;
import com.coffeeshop.tableservice.dto.BatchQrRequest;
import com.coffeeshop.tableservice.dto.CreateTableRequest;
import com.coffeeshop.tableservice.dto.UpdateTableRequest;
import com.coffeeshop.tableservice.dto.UpdateStatusRequest;
import com.coffeeshop.tableservice.entity.CoffeeTable;
import com.coffeeshop.tableservice.service.TableService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/api/tables")
@RequiredArgsConstructor
@CrossOrigin(origins = "*")
public class TableController {

    private final TableService tableService;

    @GetMapping("/health")
    public ResponseEntity<Map<String, String>> health() {
        return ResponseEntity.ok(Map.of(
                "service", "table-service",
                "status", "ok",
                "timestamp", java.time.Instant.now().toString()
        ));
    }

    @GetMapping
    public ResponseEntity<List<CoffeeTable>> findAll(@RequestParam(required = false) String branchId) {
        return ResponseEntity.ok(tableService.findAll(branchId));
    }

    @GetMapping("/{id}")
    public ResponseEntity<CoffeeTable> findById(@PathVariable String id) {
        return ResponseEntity.ok(tableService.findById(id));
    }

    @PostMapping
    public ResponseEntity<CoffeeTable> create(@Valid @RequestBody CreateTableRequest request) {
        return ResponseEntity.status(HttpStatus.CREATED).body(tableService.create(request));
    }

    @PatchMapping("/{id}")
    public ResponseEntity<CoffeeTable> update(
            @PathVariable String id,
            @Valid @RequestBody UpdateTableRequest request) {
        return ResponseEntity.ok(tableService.update(id, request));
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Map<String, Object>> delete(@PathVariable String id) {
        return ResponseEntity.ok(tableService.delete(id));
    }

    @PatchMapping("/{id}/status")
    public ResponseEntity<CoffeeTable> updateStatus(
            @PathVariable String id,
            @Valid @RequestBody UpdateStatusRequest request) {
        return ResponseEntity.ok(tableService.updateStatus(id, request.getStatus()));
    }

    @GetMapping("/{id}/qr")
    public ResponseEntity<Map<String, String>> getQrCode(@PathVariable String id) {
        String qr = tableService.getQrCode(id);
        return ResponseEntity.ok(Map.of("qrCode", qr));
    }

    @PostMapping("/qr/batch")
    public ResponseEntity<List<Map<String, String>>> getQrCodesBatch(@RequestBody(required = false) BatchQrRequest request) {
        List<String> tableIds = request != null ? request.getTableIds() : null;
        return ResponseEntity.ok(tableService.getQrBatch(tableIds));
    }

    @PostMapping("/{id}/call-staff")
    public ResponseEntity<Map<String, String>> callStaff(
            @PathVariable String id,
            @RequestBody(required = false) CallStaffRequest request) {
        return ResponseEntity.ok(tableService.callStaff(id, request));
    }
}
